### ISSUE TITLE
[Bug]: Ansible Netbox integration doesn't work anymore after upgrading to NetBox 4.0.3 from NetBox 3.6.7 

### ISSUE THREAD

**Issue by kiaz-github** [netbox-community/ansible_modules#1265]
### Ansible NetBox Collection version

v3.18.0

### Ansible version

```bash
ansible [core 2.16.7]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/chiara/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  ansible collection location = /home/chiara/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] (/usr/bin/python3)
  jinja version = 3.0.3
  libyaml = True

```

### NetBox version

v4.0.3

### Python version

3.10

### Steps to Reproduce

```
     - name: "Netbox: create/update sites"
        netbox.netbox.netbox_site:
          netbox_url: "{{ nb_url }}"
          netbox_token: "{{ nb_token }}"
          validate_certs: no
          data:
            name: "RIZ-{{ item.Kuerzel }}-{{ item.ID }}"
            status: active
            site_group: "riz"
            tenant: "RIZ"
            time_zone: "Europe/Zurich"
            physical_address: "{{ item.Hauptstandort }} \n {{ item.Adresse }} \n {{ item.PLZ }} {{ item.Ort }}"
            comments: "NSN: {{ item.Anschlussnummer }}"
          state: present
        loop: "{{ standort.list }}"
        when:
          - item.ID.startswith('10')
          - item.Status in "{{ site_valid_status_list }}"
        tags:
          - create_site
```

          This playbook worked before the upgrade.

### Expected Behavior

The connection to the NeBox API works and the Ansible Playbook is executed.

### Observed Behavior

"Failed to establish connection to NetBox API"

**Comment by LoneSnowMonkey** [comment]
I encountered this myself as well. I had upgraded my execution environment from before v3.18 release, believe we were using 3.17.0.  Recompiled to use latest (3.19.1) release of netbox ansible modules. Worked fine, but this morning I upgraded by Netbox instance from 3.7.3 to 4.0.5.  Started getting "Failed to establish connection to NetBox API", even on production playbooks that worked before upgrade.  I snapshotted before upgraded Netbox, reverted to my 3.7.3 snapshot, re-ran playbooks that failed and they worked once again.  REST calls from dev software worked fine, but in my nginx logs I am seeing 403 HTTP codes from my ansible host when playbooks are run.

One other thing odd that I noticed. I could use original tokens just fine after upgrading to netbox 4.0.5.  But any token I created after upgrade didn't work, either in playbooks, curl, or from a REST development package.  I was not able to determine cause of error, but if I put a different host into my playbook and reran playbook while doing a tcpdump I didn't see any network activity from my ansible host.

I real not really helpful, but additional info.

**Comment by LoneSnowMonkey** [comment]
The ticket I had opened about the /api 403 behavior was closed as duplicate.  The ticket it was duplicate of was closed weeks ago basically saying 'This is the intended behavior, will update documentation to say so'.

Module will need to auth all API calls.

**Comment by nicolai-hornung-bl** [comment]
Encountering the same issue - none of my playbooks work after upgrading to 4.0.3. Does someone have a workaround that doesn't involve downgrading NetBox again?

**Comment by kiaz-github** [comment]
Hi all,

I solved my issue as described here: https://github.com/netbox-community/netbox/issues/16414#issuecomment-2151805183

**Comment by jeffgdotorg** [comment]
I've just released `pynetbox` 7.3.4, which should address this bug.

**Comment by LoneSnowMonkey** [comment]
> I've just released `pynetbox` 7.3.4, which should address this bug.

Looks like its working for me now too. Thanks for your work!

**Comment by rodvand** [comment]
Fixed with pynetbox 7.3.4.
